aviary/subsystems/propulsion/utils.py

```diff
@@ -104,6 +107,54 @@ def convert_geopotential_altitude(altitude):
     return altitude
 
 
+def build_engine_deck(aviary_options: AviaryValues):
+    '''
+    Creates an EngineDeck using avaliable inputs and options in aviary_options.
+
+    Parameter
+    ----------
+    aviary_options : AviaryValues
+        Options to use in creation of EngineDecks.
+
+    Returns
+    ----------
+    engine_models : <list of EngineDecks>
+        List of EngineDecks created using provided aviary_options.
+    '''
+    # import locally placed to avoid circular import
+    from aviary.subsystems.propulsion.engine_deck import EngineDeck
+
+    # Build a single engine deck, currently ignoring vectorization
+    # of AviaryValues (use first index)
+    # TODO build test to verify this works as expected
+    engine_options = AviaryValues()
+    for entry in Aircraft.Engine.__dict__:
+        var = getattr(Aircraft.Engine, entry)
+        # check if this variable exist with useable metadata
+        try:
+            units = _MetaData[var]['units']
+            try:
+                # add value from aviary_options to engine_options
+                aviary_val = aviary_options.get_val(var, units)
+                if isinstance(aviary_val, np.ndarray):
+                    # "Convert" numpy types to standard Python types. Wrap first
+                    # index in numpy array before calling item() to safeguard against
+                    # non-standard types, such as objects
+                    aviary_val = aviary_val[0].item()
+                elif isinstance(aviary_val, (list, tuple)):
+                    aviary_val = aviary_val[0]
+                engine_options.set_val(var, aviary_val, units)
+            # if not, use default value from _MetaData
+            except KeyError:
+                # engine_options.set_val(var, _MetaData[var]['default_value'], units)
+                continue
+        except (KeyError, TypeError):
```

Comment:
A TODO, but it seems like we do this whole operation (loop over a dict, get units from metadata dict, get value + units from aviary inputs if it is in there) a few places in the code, so we should unify if we can.

Reply:
Yeah it pops up a few times. I'm not sure it's 100% the same each time but we could probably refactor around the common core there. I'll make an issue